src/util/configurations.rs

```diff
@@ -0,0 +1,83 @@
+//! This is the config struct used to pass in configurations into the channel. This is kept by the channel manager
+//! channel_limits is only passed in as a reference and only used when accepting or opening a channel
```

Comment:
What is taken as reference and what is copied seems like an implementation detail the user shouldn't need to worry about - the config gets applied, how that happen doesn't really matter to users.